plugins/outputs/nats/nats.go

```diff
@@ -85,7 +96,33 @@ func (n *NATS) Connect() error {
 
 	// try and connect
 	n.conn, err = nats.Connect(strings.Join(n.Servers, ","), opts...)
+	if n.Jetstream != nil {
+		// connect to jetstream
+		n.jetstreamClient, err = jetstream.New(n.conn)
+		if err != nil {
+			return fmt.Errorf("failed to connect to jetstream: %w", err)
+		}
+		if n.Jetstream.AutoCreateStream && !n.streamExists(n.Jetstream.StreamConfig.Name) {
+			err = choice.Check(n.Subject, n.Jetstream.StreamConfig.Subjects)
+			if err != nil {
+				return errors.New("jetstream subjects array does not contain NATS subject")
+			}
+			err = n.createStream(n.Jetstream.StreamConfig.Name)
+			if err != nil {
+				return fmt.Errorf("failed to create stream: %w", err)
+			}
+		}
+	}
+	return err
+}
+
+func (n *NATS) streamExists(stream string) bool {
+	_, err := n.jetstreamClient.Stream(context.Background(), stream)
+	return err == nil
+}
 
+func (n *NATS) createStream(stream string) error {
+	_, err := n.jetstreamClient.CreateStream(context.Background(), n.Jetstream.StreamConfig)
```

Comment:
Are there other parameters that need to be set here for this to be useful? For example, retention, discard? Or are those server side options only?

Reply:
There are many parameters which can be set to create the stream, those are defined in the https://pkg.go.dev/github.com/nats-io/nats.go/jetstream#StreamConfig 
I am using it as is, so please let me know if that is acceptable? I am taking this approach to reuse existing structs which also provides us the flexibility of using all parameters.

Reply:
I am asking to understand if we need to expose any of these settings. If this plugin option is useful to others I know we will see requests to modify options. For now this looks fine to use the defaults, but I admit I don't know nats very well.

Reply:
Currently these are all configurable, then all the options should be shown in the sample config..